src/ipc/ipc4/dai.c

```diff
@@ -115,6 +121,15 @@ int dai_config_dma_channel(struct dai_data *dd, struct comp_dev *dev, const void
 		 */
 		channel = 0;
 		break;
+	case SOF_DAI_INTEL_UAOL:
+#if defined(CONFIG_ACE_VERSION_2_0) || defined(CONFIG_ACE_VERSION_3_0)
+		channel = 0;
+		if (dai->host_dma_config[0]->pre_allocated_by_host)
+			channel = dai->host_dma_config[0]->dma_channel_id;
+#else
+		channel = copier_cfg->gtw_cfg.node_id.f.v_index;
```

Comment:
If you don't plan to enable/test UAOL for CAVS/ACE_1.0, then this line of code is dead from the beginning.

UAOL should be enabled from a know/well-supported version of hardware. pick one, don't add blanket support that will not be tested.

Reply:
Removed UAOL support on ACE1.x an Zephyr side (see [PR #69906](https://github.com/zephyrproject-rtos/zephyr/pull/69906)) and also removed here in SOF. Thanks !